pkg/apiserver/internal/v1/service.go

```diff
@@ -0,0 +1,246 @@
+	}
+	if err := unstructured.SetNestedMap(obj.Object, val, "spec"); err != nil {
+		return nil, status.Errorf(codes.Internal, fmt.Sprintf("creating services: %s", err.Error()))
+	}
+	if err := util.SetMetadata(obj, req.Spec.Metadata); err != nil {
+		return nil, status.Errorf(codes.Internal, fmt.Sprintf("creating services: %s", err.Error()))
+	}
+	if err := o.client.Create(ctx, obj); err != nil {
+		return nil, status.Errorf(codes.Internal, fmt.Sprintf("creating services: %s", err.Error()))
+	}
+	if err = o.client.Get(ctx, types.NamespacedName{
+		Name:      obj.GetName(),
+		Namespace: obj.GetNamespace(),
+	}, obj); err != nil {
+		return nil, status.Errorf(codes.Internal, fmt.Sprintf("creating service: %s", err.Error()))
+	}
+	res, err = o.convertService(obj)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, fmt.Sprintf("converting Service: %s", err.Error()))
+	}
+	return
+}
+
+func (o serviceServer) List(ctx context.Context, req *v1.ServiceListRequest) (res *v1.ServiceList, err error) {
+	var listOptions []client.ListOption
+	listOptions, err = o.validateListRequest(req)
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, err.Error())
+	}
+	obj := &unstructured.UnstructuredList{}
+	gvk := o.gvkFromService(req.Service, req.Version)
+	obj.SetGroupVersionKind(gvk)
+	if err := o.client.List(ctx, obj, listOptions...); err != nil {
+		return nil, status.Errorf(codes.Internal, fmt.Sprintf("listing services: %s", err.Error()))
+	}
+
+	res, err = o.convertServiceList(obj)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, fmt.Sprintf("converting ServiceList: %s", err.Error()))
+	}
+	return
+}
+
+func (o serviceServer) gvkFromService(service string, version string) schema.GroupVersionKind {
+	parts := strings.SplitN(service, ".", 2)
+	return schema.GroupVersionKind{
+		Kind:    parts[0],
+		Group:   parts[1],
+		Version: version,
+	}
+}
+
+func (o serviceServer) Get(ctx context.Context, req *v1.ServiceGetRequest) (res *v1.Service, err error) {
+	if err = o.validateGetRequest(req); err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, err.Error())
+	}
+	obj := &unstructured.Unstructured{}
+	gvk := o.gvkFromService(req.Service, req.Version)
+	obj.SetGroupVersionKind(gvk)
+	if err = o.client.Get(ctx, types.NamespacedName{
+		Name:      req.Name,
+		Namespace: req.Account,
+	}, obj); err != nil {
+		return nil, status.Errorf(codes.Internal, fmt.Sprintf("getting service: %s", err.Error()))
+	}
+	res, err = o.convertService(obj)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, fmt.Sprintf("converting Service: %s", err.Error()))
+	}
+	return
+}
+
+func (o serviceServer) Delete(ctx context.Context, req *v1.ServiceDeleteRequest) (*empty.Empty, error) {
+	obj := &unstructured.Unstructured{}
+	gvk := o.gvkFromService(req.Service, req.Version)
+	obj.SetGroupVersionKind(gvk)
+	if err := o.client.Get(ctx, types.NamespacedName{
```

Comment:
Why do you need to `Get` before `Delete`?

Reply:
you're right, we don't need that